Scenes/HUD/Buttons/TowerButton.gd

```diff
@@ -18,7 +18,10 @@ func _ready():
 	super._ready()
 	set_rarity(TowerProperties.get_rarity(_tower_id))
 	set_icon(TowerProperties.get_icon_texture(_tower_id))
-	set_tier_icon(_tower_id)
+	if Globals.game_mode == GameMode.enm.TOTALLY_RANDOM:
```

Comment:
I think this is supposed to include both "totally random" and "random with upgrades". So use `Globals.game_mode_is_random()`.

Reply:
No, it's supposed to use only this mode, because in "random with upgrade" we drop only towers with the first upgrade.

Reply:
Ok, I understand.